server/autolinkplugin/plugin.go

```diff
@@ -169,7 +170,7 @@ func (p *Plugin) ProcessPost(c *plugin.Context, post *model.Post) (*model.Post,
 					continue
 				}
 
-				if p.inScope(l.Scope, post.ChannelId) {
+				if p.inScope(l.Scope, channelName, teamName) {
 					newText = l.Replace(newText)
```

Comment:
`newText = l.Replace(newText)` is duplicated, can we not have it? There is no need of two `if`-s in this loop.

Reply:
I think there is, because you want to replace the text in two cases:
1. No scopes at all
2. There is a matched scope

we can merge the if conditions with an `||` statement if that's what you're suggesting?

Reply:
yes, you can merge it or let `p.inScope` decide if `newText` should be changed.